src/backend/renderer/utils/wayland.rs

```diff
@@ -68,12 +82,11 @@ pub struct Buffer {
     inner: Arc<InnerBuffer>,
 }
 
-impl From<WlBuffer> for Buffer {
-    #[inline]
-    fn from(buffer: WlBuffer) -> Self {
-        Buffer {
-            inner: Arc::new(InnerBuffer(buffer)),
-        }
```

Comment:
Removing the `From<WlBuffer>` implementation will make it impossible to construct `Buffer` outside of this crate.
As `Buffer` represents a public API used in different modules we have to provide a way constructing it.

Reply:
Yeah, I think it would be nice to have a `Buffer::from_implicit_buffer(WlBuffer)`-method as a replacement for the old `From`-implementation and perhaps a new `from_explicit_buffer`-method that could also take fences of some form.

Reply:
Ah, I didn't think about possible uses of this outside of the crate. It would be easy enough to have some kind of public constructor(s) if that's needed.

Reply:
Yeah, iirc at least `catacomb` relies on this as they are doing their own buffer management, but still use the rendering abstractions and `DrmCompositor`.